### PR TITLE
Support block-readable external plugin entries

### PR DESCRIPTION
**Pull request by ekinanp**
Supports https://github.com/puppetlabs/wash/issues/619

Signed-off-by: Enis Inan <enis.inan@puppet.com>

### CODE REVIEW

File: plugin/cache.go [hunk 1]
```diff
@@ -222,14 +222,12 @@ func cachedRead(ctx context.Context, e Entry) (entryContent, error) {
 			}
 			return newEntryContent(rawContent), nil
 		case blockReadableSignature:
-			// Go doesn't allow overloaded functions, so the external plugin entry type
```
Comment (MikaelSmith) [inline plugin/cache.go]: You decided not to keep this explanation?

Comment (ekinanp) [reply plugin/cache.go]: I moved it over to where `blockRead`'s declared

Comment (MikaelSmith) [reply plugin/cache.go]: Oh, just moved it.

**Review by MikaelSmith — COMMENTED** [review]
When do you plan to update documentation?

**Comment by ekinanp** [comment]
After this PR

**Pull request merged**